assets/js/base/components/shipping-rates-control/package.js

```diff
@@ -30,25 +33,52 @@ const Package = ( {
 				selected={ selected }
 			/>
 			{ showItems && (
-				<span>
-					{ /* @todo Show product names,
-						see: https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/1554 */ }
-					{ shippingRate.items.join( ', ' ) }
-				</span>
+				<small className="wc-block-shipping-rates-control__package-items">
+					{ Object.values( shippingRate.items ).map( ( v, i ) => {
+						const name = decodeEntities( v.name );
+						const quantity = v.quantity;
+						const displayComma =
+							i < Object.values( shippingRate.items ).length - 1;
+						return (
+							<Fragment key={ name }>
+								<Label
+									label={ `${ name } ×${ quantity }` }
+									screenReaderLabel={ sprintf(
+										// translators: %s name of the product (ie: Sunglasses), %d number of units in the current cart package
+										_n(
+											'%s (%d unit)',
+											'%s (%d units)',
+											quantity,
+											'woo-gutenberg-products-block'
+										),
+										name,
+										quantity
+									) }
+								/>
+								{ displayComma && ', ' }
```

Comment:
Just a suggestion, but since we have item names and qtys, could we do this in a list, and add commas with CSS if thats how we want to have the data appear?

Reply:
Love this idea! Done in 20b48e31a514122ebf2a714a16f1a9361233b1e3.